### PR TITLE
add check for http status

### PR DESCRIPTION
**Pull request by victor-schumacher**
**- What I did**

Add a http status check when request for stable version

**- How to verify it**

changing the stable version url for some url which don't return 200 and run ``` rit upgrade ```

**- Description for the changelog**

Add a http status check when request for stable version

### CODE REVIEW

**Review by lucasdittrichzup — APPROVED** [review]

**Review by brunasilvazup — APPROVED** [review]
You have some functional tests of teams breaking in the build, but otherwise the functionality is ok.

**Comment by victor-schumacher** [comment]
i commented my changes to test the only pipeline, and it breaks

**Comment by kaduartur** [comment]
> I commented my changes to test the only pipeline, and it breaks

The ritchie-server (https://ritchie-server.itiaws.dev) is down, try again later!

**Comment by victor-schumacher** [comment]
> > I commented my changes to test the only pipeline, and it breaks
> 
> The ritchie-server (https://ritchie-server.itiaws.dev) is down, try again later!
Hi @kaduartur, i tried now but still failing, is there any specific time?

**Comment by victor-schumacher** [comment]
> > > I commented my changes to test the only pipeline, and it breaks
> > 
> > 
> > The ritchie-server (https://ritchie-server.itiaws.dev) is down, try again later!
> > Hi @kaduartur, i tried now but still failing, is there any specific time?

Since the QA environment used on team's functional test was removed, we are removing this step from pipeline too.

File: pkg/version/version.go [hunk 1]
```diff
@@ -72,10 +72,10 @@ func requestStableVersion(stableVersionUrl string, httpClient *http.Client) (str
 	}
 
 	response, err := httpClient.Do(request)
-
-	if err != nil {
+	if err != nil || response.StatusCode != http.StatusOK {
```
Comment (kaduartur) [inline pkg/version/version.go:75]: I think that when StatusCode isn't 200 the err can be nil and return this not make sense. 

Could you create a new error type and create a new test for this case?

**Review by kaduartur — CHANGES_REQUESTED** [review]

File: pkg/version/version.go (hunk 1 above)
Comment (victor-schumacher) [reply pkg/version/version.go]: Thanks for the advice, that's so true.
Done.

**Review by kaduartur — APPROVED** [review]

**Pull request merged**